Enabled beaker provisioning for pytest-ansible tests

cpinjani opened the pull request:
Signed-off-by: Chandan Pinjani <cpinjani@redhat.com>

cpinjani commented:
If PSI provisioning fails, this enables beaker provisioning by default.

Tested by running pipeline over forked branch:
PSI: https://gitlab.com/dpunia/pki/-/pipelines/312408724
Beaker: https://gitlab.com/dpunia/pki/-/pipelines/312531159

dpuniaredhat commented:
@cpinjani 1 point, Can we also add the parameter "provision_beaker : true" to run the jobs only on beaker because as observation from last openstack issue sometime machine inaccessible after running few test or packet loss which cause machine hang in between after provisioning on openstack.

We can make condition like.
1. provision_beaker : false it will first try on openstack in-case failure next try out on beaker provisioning. 
2. provision_beaker : true only go for beaker provisioning. 

Rest LGTM

edewata commented on tests/dogtag/pytest-ansible/.gitlab-ci.yml:
```diff
@@ -1,4 +1,5 @@
-image: bbhavsar/dogtagpki-test-essentials-container
+#image: bbhavsar/dogtagpki-test-essentials-container
+image: docker-registry.upshift.redhat.com/pki/dogtagpki-test-essentials-container
```
Is this image available publicly?

cpinjani replied:
No, its in Red Hat registry

edewata replied:
Does the image contain any sensitive information? If not, could we put it https://quay.io/organization/dogtagpki ?

cpinjani replied:
Actually yes, its has information pertaining to certs/keytab files for authenticating to Beaker

cpinjani commented:
> @cpinjani 1 point, Can we also add the parameter "provision_beaker : true" to run the jobs only on beaker because as observation from last openstack issue sometime machine inaccessible after running few test or packet loss which cause machine hang in between after provisioning on openstack.
> 
> We can make condition like.
> 
> 1. provision_beaker : false it will first try on openstack in-case failure next try out on beaker provisioning.
> 2. provision_beaker : true only go for beaker provisioning.
> 
> Rest LGTM

Implemented this. Added 'provision_beaker' as pipeline variable.
If false: it tries first on openstack, in-case of failure next try out on beaker provisioning
If true: Run all jobs using beaker provisioning

provision_beaker - false: https://gitlab.com/dpunia/pki/-/jobs/1314278941
provision_beaker - false, PSI Down: https://gitlab.com/dpunia/pki/-/jobs/1311912037
provision_beaker - true: https://gitlab.com/dpunia/pki/-/jobs/1311560825

dpuniaredhat commented on tests/dogtag/pytest-ansible/.gitlab-ci.yml:
```diff
@@ -30,255 +32,274 @@ before_script:
-    - py.test --html=$CI_PROJECT_DIR/${CI_JOB_NAME}_html_report.html --self-contained-html --ansible-user=fedora --ansible-become --ansible-inventory $HOSTFILE --ansible-host-pattern master $PYTEST_DIR/pytest/sanity/test_role_users.py -q -s --junitxml $CI_PROJECT_DIR/role-user-creation_junit.xml -vv
+    - ansible-playbook -i $HOSTFILE $PYTEST_DIR/installation/main.yml --extra-vars "topology=topology-01" -vv  | sed 's/\\n/\n/g'
+    - py.test --html=$CI_PROJECT_DIR/${CI_JOB_NAME}_html_report.html --self-contained-html --ansible-inventory $HOSTFILE --ansible-host-pattern master $PYTEST_DIR/pytest/sanity/test_role_users.py -q -s --junitxml $CI_PROJECT_DIR/role-user-creation_junit.xml -vv
+  only:
```
@cpinjani remove only before merge

06shalini commented:
LGTM
